js/samples/react-sample-simple/src/AppOptions.js

```diff
@@ -0,0 +1,151 @@
+  const [language, setLanguage] = useState('en');
+
+  const styles = {
+    parent_frame: {
+        height: 400
+    },
+    section: {
+        border: "1px solid #cccccc",
+        borderRadius: "4px",
+        boxSizing: "border-box",
+        display: "block",
+        marginBottom: "20px",
+        padding: "20px 20px 10px 20px",
+        position: "relative",
+        width: "100%"
+    },
+    immersive_reader_button: {
+        backgroundColor: "white",
+        marginTop: "5px",
+        border: "1px solid black",
+        float: "right"
+    }
+  };
+
+  const getCredentials = async () => {
+
+    // Form details to be passed to fetch
+    const details = {
+      grant_type: 'client_credentials',
+      client_id: process.env.REACT_APP_CLIENT_ID,
+      client_secret: process.env.REACT_APP_CLIENT_SECRET,
+      resource: 'https://cognitiveservices.azure.com/'
+    };
+
+    // Build up the form data -> it needs to be converted to a form type
+    let formBodyArr = [];
+    for (var property in details) {
+      var encodedKey = encodeURIComponent(property);
+      var encodedValue = encodeURIComponent(details[property]);
+      formBodyArr.push(encodedKey + "=" + encodedValue);
+    }
+
+    const formBodyStr = formBodyArr.join("&"); // This is what we can pass to the post request
+
+    try {
+
+      const response = await fetch(`http://localhost:3000/${process.env.REACT_APP_TENANT_ID}/oauth2/token`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+        body: formBodyStr,
+      });
+      const json = await response.json();
+      const { access_token } = json;
+      setToken(access_token)
+
+    }
+    catch (err) {
+      alert('There was a problem fetching your credentials, please check the console and make sure your environment variables are prefixed with REACT_APP_');
+      console.log({ err })
+    }
+  }
+
+  const handleLaunchImmersiveReader = async (sampleId) => {
+
+    var langElement = document.getElementById('lang_' + sampleId);
+    var lang = sampleId === 'DisableLanguageDetection' ? langElement.value : 'en';
+
+    const data = {
+        title: document.getElementById('title' + sampleId).innerText,
+        chunks: [{
+            content: document.getElementById('text' + sampleId).innerHTML,
+            lang: lang
+        }]
+    };
+
+    // Learn more about options https://docs.microsoft.com/azure/cognitive-services/immersive-reader/reference#options
+    let options = {};
+
+    options.disableLanguageDetection = sampleId === 'DisableLanguageDetection';
+    options.disableGrammar = sampleId === 'DisableGrammar';
+    options.disableTranslation = sampleId === 'DisableTranslation';
+    options.parent = sampleId === 'Parent' && document.getElementById("checkboxParent").checked ? document.getElementById('parentDiv') : null;
+
+    try {
+      await launchAsync(token, process.env.REACT_APP_SUBDOMAIN, data, options)
```

Comment:
REACT_APP_SUBDOMAIN

check for this up front with explicit error msg

Reply:
done, checking values at getCredentials in constructor